src/schemas/generator/v0/index.ts

```diff
@@ -21,12 +21,12 @@ export type GeneratorSchema = z.infer<typeof GeneratorFileDataSchema>
 export function migrate(generator: GeneratorSchema): v1.GeneratorSchema {
   return {
     version: '1.0',
-    allowlist: generator.allowlist,
-    includeStaticAssets: generator.includeStaticAssets,
-    options: generator.options,
     recordingPath: generator.recordingPath,
+    options: generator.options,
+    testData: generator.testData,
     rules: generator.rules,
+    allowlist: generator.allowlist,
+    includeStaticAssets: generator.includeStaticAssets,
```

Comment:
The issue was actually related to the order of the fields in the migration, which caused the `JSON.stringify` comparison to return `false`. I made a ticket to address this behaviour https://github.com/grafana/k6-studio/issues/403